### ISSUE TITLE
Dual Hosted Tutorial Content

### ISSUE THREAD

**Issue by thecrudge** [kubernetes/website#15787]
**Problem:**
Currently, the katacoda tutorials are embedded in the kubernetes website. The majority of the issues that get opened toward those tutorials are support requests, rather than issues against the documentation. Currently, these tutorials are also hosted elsewhere in the org, and this has raised the question about how to handle dual hosted content. 

**Proposed Solution:**
Lazy consensus is that we remove the katacoda tutorials and link to them somewhere in the in the docs. Possibly the top level page that explains the tutorial.

**Page(s) to Update:**
https://kubernetes.io/docs/tutorials/kubernetes-basics/create-cluster/cluster-intro/
 -> https://kubernetes.io/docs/tutorials/kubernetes-basics/create-cluster/cluster-interactive/

https://kubernetes.io/docs/tutorials/kubernetes-basics/deploy-app/deploy-intro/
 -> https://kubernetes.io/docs/tutorials/kubernetes-basics/deploy-app/deploy-interactive/

https://kubernetes.io/docs/tutorials/kubernetes-basics/explore/explore-intro/
 -> https://kubernetes.io/docs/tutorials/kubernetes-basics/explore/explore-interactive/

https://kubernetes.io/docs/tutorials/kubernetes-basics/expose/expose-intro/
 -> https://kubernetes.io/docs/tutorials/kubernetes-basics/expose/expose-interactive/

https://kubernetes.io/docs/tutorials/kubernetes-basics/scale/scale-intro/
 -> https://kubernetes.io/docs/tutorials/kubernetes-basics/scale/scale-interactive/

https://kubernetes.io/docs/tutorials/kubernetes-basics/update/update-intro/
 -> https://kubernetes.io/docs/tutorials/kubernetes-basics/update/update-interactive/

**Comment by bradtopol** [comment]
So I thought the issue is we have a link like the following:
https://kubernetes.io/docs/tutorials/kubernetes-basics/create-cluster/cluster-interactive/
which is exactly the same as this:
https://www.katacoda.com/courses/kubernetes/launch-single-node-cluster

and that we should just instead link to the katacoda version so folks understand that this is Katacoda content and if they have any problems with the katacoda link and associated tutorial they should send their issues to Katacoda instead of mistakenly sending the issue to us because they originally used one of our links (i.e. https://kubernetes.io/docs/tutorials/kubernetes-basics/create-cluster/cluster-interactive/) to start the tutorial.  
Am I understanding the issue properly?

If so then the work item is to find all our links that are just our version of the katacoda links and then replace our links with the corresponding Katacoda link.  Also above the link we should describe what tutorial is run by the link, that it is hosted by Katacoda, and any issues with running the tutorial should be directed to the Katacoda support team.

**Comment by thecrudge** [comment]
@bradtopol 

That is the exact issue. I was only suggesting we link to it the top level page, for those interactive links.

**Comment by BenHall** [comment]
I just went over the initial tickets I could find to do some initial triage and responded. I believe a number of issues can be closed. We will continue to monitor and respond accordingly. 

The main issue I wasn't aware of was #12968 and the impact it was having. This was related to the hello-minikube content. While the content mentions the solution, people just clicked the link, see the error and raise a ticket.

We are now showing a friendly error message which will hopefully stop users raising issues.

**Comment by steveperry-53** [comment]
These are the two places I know of where you can view and interact with
this content:

Kubernetes Bootcamp
<https://kubernetesbootcamp.github.io/kubernetes-bootcamp/index.html>
Kubernetes Basics <https://kubernetes.io/docs/tutorials/kubernetes-basics/>

Apparently, the two sites have fallen out of sync.

In my mind, this is the fundamental question:
Is the Kubernetes Bootcamp
<https://kubernetesbootcamp.github.io/kubernetes-bootcamp/index.html> site
going to be maintained?

If yes, then we should take down the Kubernetes Basics pages, and instead
provide a link to Kubernetes Bootcamp.

If no, then we should take down the Kubernetes Bootcamp site.

**Comment by zacharysarah** [comment]
@BenHall 👋 Thanks for sharing your observations. 

I echo @steveperry-53's fundamental concern. Dual-sourced content isn't acceptable. Tutorials need to reside at one URL or the other.

The Kubernetes Basics tutorial ranks consistently in or near the website's ten most-visited pages. The future of tutorial content is an important issue.

## Support gap

SIG Docs experiences a support gap with Katacoda. SIG Docs maintains a website that [hosts dual-sourced content](https://kubernetesbootcamp.github.io/kubernetes-bootcamp/index.html), records tutorial issues and triages them, but receives only intermittent support from Katacoda, without any content visibility or service level agreement (SLA) about when or how issues are monitored or fixed.

I appreciate your good grace in responding to and addressing concerns for the past two years, @BenHall, but the way things have been can't be the way they continue. We need a sustainable model that includes clear expectations of support, timeliness, and communication.

> We move the content into the main repository. This will allow content changes and improvements to be more in a consistent way.

Frankly, I'm skeptical.

Who does Katacoda propose will maintain this content? Will it just be you, @BenHall? (Would you ever be able to take a vacation again? 😳) Is Katacoda willing to field a dedicated headcount for tutorial content, with the understanding that ongoing hosting is tied to a minimum level of staffing? Is Katacoda willing to explicitly state this?

Unless Katacoda guarantees headcount for support, transferring content into k/website only shifts maintenance with unclear expectations onto an overburdened community. Without dedicated support, my answer is no.

> The use of Kubectl Proxy seems to confuse people. When the proxy isn't running, they get an unhelpful error and raise the ticket.
...
> Alternatively, we can also look at the platform ensuring users don't try to run the cURL command before it's been started.

These examples and [this comment](https://github.com/kubernetes/website/issues/15787#issuecomment-520763797) illustrate the support gap named above. ☝️ 

> Maybe we can setup a extension to the bot which automatically assigns the issues to us if they have been created via one of the interactive pages?

A CI integration is pretty but useless without an SLA that explicitly states who fixes tutorial issues, when they'll be fixed, and how communication happens. Automation _per se_ solves none of these concerns.

> It would be great to be able to close issues that have been fixed or related to another issue (such as above).

If Katacoda provides dedicated staffing with an SLA, it makes sense to extend approver privileges for k/website to that staff.

## TLDR

Tutorial content is valuable, but it requires dedicated, explicitly-stated support to continue at kubernetes.io.

Resolve dual-sourced [Bootcamp content](https://kubernetesbootcamp.github.io/kubernetes-bootcamp/index.html) by removing it at one site or the other.

CC: @jimangel @Bradamant3 @kbarnard10 @jaredbhatti

**Comment by zacharysarah** [comment]
```
## Lazy consensus 

I want to make sure this issue gets adequate discussion, but I also want to make sure there's clear expectations about what happens by default. I also hope it's never something we have to act on.

For Katacoda tutorials to continue residing at kubernetes.io/docs/tutorials, the following must be true by 15 September 2019:
- [ ] Katacoda provides a clear SLA to SIG Docs chairs with minimum levels of staffing, response times, and fix times. 
- [ ] SIG Docs chairs agree on SLA provisions.
- [ ] Content is single-sourced. The [Kubernetes Bootcamp](https://kubernetesbootcamp.github.io/kubernetes-bootcamp/index.html) site is taken down and 301-redirects to the relevant tutorials on kubernetes.io.

Otherwise, SIG Docs will remove tutorial content at its discretion. SIG Docs may provide 301 redirects to Katacoda URLs at the best of its ability, but will not assume responsibility for ensuring correct redirects.

**Comment by BenHall** [comment]
To provide an update on the dual content issue. I've investigate the Kubernetes Bootcamp and refreshed my memory. Last year I did make changes to the website bootcamp to automatically redirect to the Kubernetes Docs in an attempt to close it so we didn't have this issue. However, it looks like there are two repos (unsure why) and the website wasn't pointing that the intended repository. 

I have now sent a PR to the original owners (https://github.com/kubernetesbootcamp/kubernetes-bootcamp/pull/6) and also a Twitter message. Once merged, the content will hopefully redirect accordingly. 

I don't believe anyone on this thread has permissions to the repo so I have also added a redirect so the interactive content is never displayed and users are alerted where to go instead as shown below.

![Screenshot 2019-08-21 08 38 10](https://user-images.githubusercontent.com/82614/63412854-77bdb700-c3f0-11e9-84fd-07ecf9df7f81.png)

The interactive section of the content is no longer dual hosted and dedicated to Kubernetes.io.

**Comment by jocatalin** [comment]
Hi, I'm the content creator of the original bootcamp tutorial. AFAIR  @pwittrock created the [kubernetesbootcamp.github.io](https://github.com/kubernetesbootcamp/) organisation as an intermediary step, before having the code merged into the official k8s website. @pwittrock can you please merge @BenHall 's [PR](https://github.com/kubernetesbootcamp/kubernetes-bootcamp/pull/6) and eventually archive/disable the GitHub branch publishing for [kubernetesbootcamp.github.io](https://github.com/kubernetesbootcamp/) I don't have rights anymore on that organization.
